### ISSUE TITLE
Is there any plan to support custom markdown directives?

### ISSUE THREAD

**Issue by tonyalaribe** [raphlinus/pulldown-cmark#616]
I have a usecase where we need to support markdown directives. Is there support for this? Or is it something I could contribute? 

This is what I mean: 
https://talk.commonmark.org/t/generic-directives-plugins-syntax/444

The idea being that we could define and plug in this new directives into the AST and define custom renderers for it.

**Comment by ssokolow** [comment]
\*nod* I've been working around this by using custom HTML tags and running a second pass over the output using quick-xml, configured to allow unbalanced tags.

**Comment by Martin1887** [comment]
In the short term they will not be supported. In the long term I would like to have an interface for pluggable rules from other crates, and this could be implemented in another crate.

**Comment by kamoshi** [comment]
This would be extremely useful to have, as it would allow arbitrary custom extensions to Markdown. The use case I have in mind is a website generator where people often want to add custom elements embedded in articles written in Markdown.

**Comment by LeonardoTan19** [comment]
Agreed. This is the foundation for custom Markdown grammars and plugins.
